Jenkinsfile

```diff
@@ -48,6 +49,79 @@ pipeline {
 
     stage('Tests') {
       parallel {
+        stage('Prepare Windows 2016 with containers') {
+          agent { label 'executor-windows-2016-containers' }
+          stages {
+            stage('Configure Windows Node'){
+              steps {
+                powershell '.\\expose-daemon.ps1'
+                script {
+                  env.WINDOWS_IP = powershell(returnStdout: true, script:  '(curl http://169.254.169.254/latest/meta-data/local-ipv4).Content').trim()
+                  env.WINDOWS_DOCKER_CERT_CA = powershell(returnStdout: true, script:  'cat $env:USERPROFILE\\.docker\\ca.pem')
+                  env.WINDOWS_DOCKER_CERT_CERT = powershell(returnStdout: true, script:  'cat $env:USERPROFILE\\.docker\\cert.pem')
+                  env.WINDOWS_DOCKER_CERT_KEY = powershell(returnStdout: true, script:  'cat $env:USERPROFILE\\.docker\\key.pem')
+                  env.WINDOWS_READY = true
+                }
+              }
+            }
+            stage('Wait for Main Node') {
+              steps {
+                waitUntil {
+                  script {
+                    return (env.MAIN_NODE_DONE == "true");
+                  }
+                }
+              }
+            }
+          }
+        }
+
+        stage('With Windows 2016 containers') {
+          stages {
+            stage("Wait for Windows node") {
+              steps {
+                waitUntil {
+                  script {
+                    return (env.WINDOWS_READY == "true");
+                  }
+                }
+                script {
+                  env.WINDOWS_DOCKER_CERT_DIR = "${pwd()}/tmp-docker"
+                }
+
+                sh "mkdir ${env.WINDOWS_DOCKER_CERT_DIR}"
+                writeFile file: "${env.WINDOWS_DOCKER_CERT_DIR}/ca.pem", text: env.WINDOWS_DOCKER_CERT_CA
+                writeFile file: "${env.WINDOWS_DOCKER_CERT_DIR}/cert.pem", text: env.WINDOWS_DOCKER_CERT_CERT
+                writeFile file: "${env.WINDOWS_DOCKER_CERT_DIR}/key.pem", text: env.WINDOWS_DOCKER_CERT_KEY
+              }
+            }
+
+// Integration tests
```

Comment:
Needs indenting

Reply:
The comments aren't actually necessary. I'd added as a visual aid so that as you're scrolling down you can quite easily see where things are. I think indenting might defeat the purpose.

Reply:
The comment should match the indent block that it's describing. I think by doing a level-0 indent it's making it hard to see that this just encompases the integration tests. I'd almost maybe just remove them wholesale if you can make the `E2E - Puppet 6 - Conjur 5` step be more descriptive that it's an integration step.